src/code/z_play.c

```diff
@@ -110,34 +110,253 @@
+ *
+ * @param[in] gameState GameState, promoted to globalCtx inside.
+ * @param[in] actor Actor to test.
+ * @param[out] yaw Facing angle of the actor, or reverse if in the back room.
+ * @return true if \p actor is a door and the sides are in different rooms, false otherwise
+ */
+s32 func_8016A02C(GameState* gameState, Actor* actor, s16* yaw) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    TransitionActorEntry* transitionActor;
+    s8 frontRoom;
+
+    if (actor->category != ACTORCAT_DOOR) {
+        return false;
+    }
+
+    transitionActor = &globalCtx->doorCtx.transitionActorList[(u16)actor->params >> 10];
+    frontRoom = transitionActor->sides[0].room;
+    if (frontRoom == transitionActor->sides[1].room) {
+        return false;
+    }
+
+    if (frontRoom == actor->room) {
+        *yaw = actor->shape.rot.y;
+    } else {
+        *yaw = actor->shape.rot.y + 0x8000;
+    }
+
+    return true;
+}
+
+// Unused
+/**
+ * @brief Tests if \p pos is underwater.
+ *
+ * @param[in] globalCtx GlobalContext
+ * @param[in] pos position to test
+ * @return true if inside a waterbox and not above a void.
+ */
+s32 Play_IsUnderwater(GlobalContext* globalCtx, Vec3f* pos) {
+    WaterBox* waterBox;
+    CollisionPoly* poly;
+    Vec3f waterSurfacePos;
+    s32 bgId;
+
+    waterSurfacePos = *pos;
+
+    if ((WaterBox_GetSurface1(globalCtx, &globalCtx->colCtx, waterSurfacePos.x, waterSurfacePos.z, &waterSurfacePos.y,
+                              &waterBox) == true) &&
+        (pos->y < waterSurfacePos.y) &&
+        (BgCheck_EntityRaycastFloor3(&globalCtx->colCtx, &poly, &bgId, &waterSurfacePos) != BGCHECK_Y_MIN)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+// z_demo and EnTest4
+// This data appears to be a boolean. It is only set by Play_Init.
+s32 func_8016A168(void) {
+    return D_801D0D50;
+}
+
+// List of cutscene numbers.
+extern s16 D_801D0D64[];
+// s16 D_801D0D64[] = { -3, -2, -4, -5, -7, -11, -8, -9, -6, -16 };
+
+// Used by Player
+/**
+ * Extract the common actor cutscene ids from a scene and set the cutscene ids in globalCtx->playerActorCsIds.
+ * Set to -1 by default. If there is an ActorCutscene where csCamSceneDataId matches the appropriate element of
+ * D_801D0D64, set the corresponding playerActorCsId (and possibly change its priority for the zeroth one)
+ */
+void Play_AssignPlayerActorCsIdsFromScene(GameState* gameState, s32 cutscene) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    s32 i;
+    s16* phi_s3 = globalCtx->playerActorCsIds;
+    s16* phi_s1 = D_801D0D64;
+
+    for (i = 0; i < ARRAY_COUNT(globalCtx->playerActorCsIds); i++, phi_s3++, phi_s1++) {
+        ActorCutscene* actorCutscene;
+        s32 currCutscene;
+
+        *phi_s3 = -1;
+
+        for (currCutscene = cutscene; currCutscene != -1; currCutscene = actorCutscene->additionalCutscene) {
+            actorCutscene = ActorCutscene_GetCutscene(currCutscene);
```

Comment:
`actorCutscene` is a pointer to the cutscene, while `currCutscene` and `cutscene` are `Id`'s tied to each cutscene, and I think this distinction should be made clear. Futhermore, I believe the convention is to use `cur` with 1 `r` instead of two. So:
`currCutscene` -> `curActorCsId`
`cutscene` -> `startActorCsId`
`phi_s3` -> `curPlayerActorCsId` or `playerActorCsId`

It's also okay to leave it outside the scope of this PR, but I would support either
`additionalCutscene` -> `additionalActorCsId` or `additionalCutscene` -> `additionalCsId` perhaps?

Reply:
Or also using `next` instead of `additional`, I think `next` is more common?

Reply:
If it's a linked list, I think `next` is standard?

Reply:
From my understanding yeah